### PR TITLE
Support some external status check api

### PR DESCRIPTION
**Pull request by epropulsion-tech**

### CODE REVIEW

File: event_webhook_types.go [hunk 1]
```diff
@@ -189,6 +189,7 @@ type IssueCommentEvent struct {
 		NoteableType string  `json:"noteable_type"`
 		AuthorID     int     `json:"author_id"`
 		CreatedAt    string  `json:"created_at"`
+		DiscussionID string  `json:"discussion_id"`
```
Comment (svanharmelen) [inline event_webhook_types.go:192]: I cannot find this field in the docs. Are you sure it's there? And additionally we try to keep the order inline with the order of the docs (or the GitLab source code), so is this the correct location for this field?

File: external_status_checks.go [hunk 2]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
 
 	return pscs, resp, err
 }
+
+type SetExternalStatusCheckStatusOptions struct {
+	SHA                   *string `url:"sha" json:"sha"`
+	ExternalStatusCheckID *int    `url:"external_status_check_id" json:"external_status_check_id"`
+	Status                *string `url:"status,omitempty" json:"status,omitempty"`
+}
```
Comment (svanharmelen) [inline external_status_checks.go:96]: Please all a header (comment) and add missing `omitempty` tag options (should be on all fields).

File: external_status_checks.go [hunk 3]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+
+// SetExternalStatusCheckStatus set status of an external status check
```
Comment (svanharmelen) [inline external_status_checks.go:98]:
```suggestion
// SetExternalStatusCheckStatus sets the status of an external status check.
```

File: external_status_checks.go [hunk 4]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+//
+// Gitlab API docs:
+// https://docs.gitlab.com/ee/api/status_checks.html#set-status-of-an-external-status-check
+func (s *ExternalStatusChecksService) SetExternalStatusCheckStatus(pid interface{}, mergeRequestIID int, opt *SetExternalStatusCheckStatusOptions, options ...RequestOptionFunc) (*Response, error) {
```
Comment (svanharmelen) [inline external_status_checks.go:102]:
```suggestion
func (s *ExternalStatusChecksService) SetExternalStatusCheckStatus(pid interface{}, mergeRequest int, opt *SetExternalStatusCheckStatusOptions, options ...RequestOptionFunc) (*Response, error) {
```

File: external_status_checks.go [hunk 5]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, err
+	}
+	u := fmt.Sprintf("projects/%s/merge_requests/%d/status_check_responses", PathEscape(project), mergeRequestIID)
```
Comment (svanharmelen) [inline external_status_checks.go:107]:
```suggestion
	u := fmt.Sprintf("projects/%s/merge_requests/%d/status_check_responses", PathEscape(project), mergeRequest)
```

File: external_status_checks.go (hunk 4 above)
Comment (svanharmelen) [inline external_status_checks.go:102]: Following the order in the docs, this method should be defined below `ListMergeStatusChecks`

File: external_status_checks.go [hunk 6]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+//
+// Gitlab API docs:
+// https://docs.gitlab.com/ee/api/status_checks.html#update-external-status-check
+func (s *ExternalStatusChecksService) UpdateExternalStatusCheck(pid interface{}, checkID int, opt *UpdateExternalStatusCheckOptions, options ...RequestOptionFunc) (*Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("projects/%s/external_status_checks/%d", PathEscape(project), checkID)
```
Comment (svanharmelen) [inline external_status_checks.go:178]:
```suggestion
	}
	u := fmt.Sprintf("projects/%s/external_status_checks/%d", PathEscape(project), check)
```

File: external_status_checks.go [hunk 7]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+	Name               *string `url:"name,omitempty" json:"name,omitempty"`
+	ExternalURL        *string `url:"external_url,omitempty" json:"external_url,omitempty"`
+	ProtectedBranchIDs *[]int  `url:"protected_branch_ids,omitempty" json:"protected_branch_ids,omitempty"`
+}
+
+// UpdateExternalStatusCheck update external status check
+//
+// Gitlab API docs:
+// https://docs.gitlab.com/ee/api/status_checks.html#update-external-status-check
+func (s *ExternalStatusChecksService) UpdateExternalStatusCheck(pid interface{}, checkID int, opt *UpdateExternalStatusCheckOptions, options ...RequestOptionFunc) (*Response, error) {
```
Comment (svanharmelen) [inline external_status_checks.go:172]:
```suggestion
func (s *ExternalStatusChecksService) UpdateExternalStatusCheck(pid interface{}, check int, opt *UpdateExternalStatusCheckOptions, options ...RequestOptionFunc) (*Response, error) {
```

File: external_status_checks.go [hunk 8]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+
+	req, err := s.client.NewRequest(http.MethodDelete, u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
+
+type UpdateExternalStatusCheckOptions struct {
```
Comment (svanharmelen) [inline external_status_checks.go:162]: Please add a header...

File: external_status_checks.go [hunk 9]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+	Name               *string `url:"name,omitempty" json:"name,omitempty"`
+	ExternalURL        *string `url:"external_url,omitempty" json:"external_url,omitempty"`
+	ProtectedBranchIDs *[]int  `url:"protected_branch_ids,omitempty" json:"protected_branch_ids,omitempty"`
+}
+
+// UpdateExternalStatusCheck update external status check
```
Comment (svanharmelen) [inline external_status_checks.go:168]:
```suggestion
// UpdateExternalStatusCheck updates an external status check.
```

File: external_status_checks.go [hunk 10]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+
+	req, err := s.client.NewRequest(http.MethodPost, u, opt, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
+
+// DeleteExternalStatusCheck delete external status check
```
Comment (svanharmelen) [inline external_status_checks.go:142]:
```suggestion
// DeleteExternalStatusCheck deletes an  external status check.
```

File: external_status_checks.go [hunk 11]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+//
+// Gitlab API docs:
+// https://docs.gitlab.com/ee/api/status_checks.html#delete-external-status-check
+func (s *ExternalStatusChecksService) DeleteExternalStatusCheck(pid interface{}, checkID int, options ...RequestOptionFunc) (*Response, error) {
```
Comment (svanharmelen) [inline external_status_checks.go:146]:
```suggestion
func (s *ExternalStatusChecksService) DeleteExternalStatusCheck(pid interface{}, check int, options ...RequestOptionFunc) (*Response, error) {
```

File: external_status_checks.go [hunk 12]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, err
+	}
+
+	u := fmt.Sprintf("projects/%s/external_status_checks/%d", PathEscape(project), checkID)
```
Comment (svanharmelen) [inline external_status_checks.go:152]:
```suggestion
	}
	u := fmt.Sprintf("projects/%s/external_status_checks/%d", PathEscape(project), check)
```

File: external_status_checks.go [hunk 13]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+
+	req, err := s.client.NewRequest(http.MethodPost, u, opt, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.client.Do(req, nil)
+}
+
+type CreateExternalStatusCheckOptions struct {
```
Comment (svanharmelen) [inline external_status_checks.go:117]: Please add a header

File: external_status_checks.go [hunk 14]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+	Name               *string `url:"name" json:"name"`
+	ExternalURL        *string `url:"external_url" json:"external_url"`
+	ProtectedBranchIDs *[]int  `url:"protected_branch_ids,omitempty" json:"protected_branch_ids,omitempty"`
```
Comment (svanharmelen) [inline external_status_checks.go:120]: Add missing `omitempty` options

File: external_status_checks.go [hunk 15]
```diff
@@ -88,3 +88,99 @@ func (s *ExternalStatusChecksService) ListProjectStatusChecks(pid interface{}, o
...
+}
+
+// CreateExternalStatusCheck create external status check
```
Comment (svanharmelen) [inline external_status_checks.go:123]:
```suggestion
// CreateExternalStatusCheck creates an external status check.
```

**Review by svanharmelen — CHANGES_REQUESTED** [review]

File: event_webhook_types.go (hunk 1 above)
Comment (epropulsion-tech) [reply event_webhook_types.go:192]: I'm sure it's there, and it's correct order

3gpzCA7CjB

**Review by svanharmelen — APPROVED** [review]
Made a few last tweaks, looks good now. Thanks 👍🏻

**Pull request merged**